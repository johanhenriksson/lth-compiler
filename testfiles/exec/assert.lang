int main() {
    assert(1 == 1);
    assert(1 != 0);
    assert(0 < 1);
    assert(1 > 0);
    assert(1 >= 0);
    assert(0 >= 0);
    assert(1 <= 1);
    assert(0 <= 1);

    assert(true);
    assert(true || true);
    assert(true || false);
    assert(false || true);
    assert(true && true);

    except(false);
    except(false || false);
    except(true && false);
    except(false && true);
    except(false && false);

    return 0;
}
