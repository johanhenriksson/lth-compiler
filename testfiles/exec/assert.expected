; Library Functions
declare void @print(i64)
declare i64 @read()
declare i64 @truncate(double)
declare double @promote(i64)
declare void @assert(i1)

define i64 @main() {
%tmp.3 = icmp eq i1 1, 1
call void @assert(i1 %tmp.3)
%tmp.4 = icmp ne i1 1, 0
call void @assert(i1 %tmp.4)
%tmp.5 = icmp sgt i1 1, 0
call void @assert(i1 %tmp.5)
%tmp.6 = icmp slt i1 0, 1
call void @assert(i1 %tmp.6)
%tmp.7 = icmp sge i1 1, 0
call void @assert(i1 %tmp.7)
%tmp.8 = icmp sge i1 0, 0
call void @assert(i1 %tmp.8)
%tmp.9 = icmp sle i1 1, 1
call void @assert(i1 %tmp.9)
%tmp.10 = icmp sle i1 0, 1
call void @assert(i1 %tmp.10)
ret i64 0
}

